docs/site/migration/express-middleware.md

```diff
@@ -6,8 +6,251 @@ sidebar: lb4_sidebar
+      "limit": "100kb"
+    },
+    "cors": false
+  }
+}
+```
+
+We will be using a base Express app for mounting the LB4 app. Notice how we are
+loading the middleware from the LB3 in this app. These middleware will be shared
+by both LB3 and LB4 apps. We are not adding the `loopback.rest` middleware, it
+will be loaded by the `Lb3AppBooterComponent`.
+
+{% include code-caption.html content="lb3app/server.ts" %}
+
+```ts
+import {ApplicationConfig} from '@loopback/core';
+import * as express from 'express';
+import {Request, Response} from 'express';
+import * as http from 'http';
+import pEvent from 'p-event';
+import * as path from 'path';
+import {CoffeeShopApplication} from './application';
+
+const loopback = require('loopback');
+const compression = require('compression');
+const cors = require('cors');
+const helmet = require('helmet');
+
+export class ExpressServer {
+  private app: express.Application;
+  public readonly lbApp: CoffeeShopApplication;
+  private server?: http.Server;
+
+  constructor(options: ApplicationConfig = {}) {
+    this.app = express();
+    this.lbApp = new CoffeeShopApplication(options);
+
+    // Middleware for LB3
+    this.app.use(loopback.favicon());
+    this.app.use(compression());
+    this.app.use(cors());
+    this.app.use(helmet());
+
+    // Mount the LB4 REST API
+    this.app.use('/api', this.lbApp.requestHandler);
+
+    // Custom Express routes
+    this.app.get('/', function(_req: Request, res: Response) {
+      res.sendFile(path.join(__dirname, '../public/express.html'));
+    });
+
+    this.app.get('/hello', function(_req: Request, res: Response) {
+      res.send('Hello world!');
+    });
+
+    // Serve static files in the public folder
+    this.app.use(express.static(path.join(__dirname, '../public')));
+  }
+
+  public async boot() {
+    await this.lbApp.boot();
+  }
+
+  public async start() {
+    await this.lbApp.start();
+    const port = this.lbApp.restServer.config.port || 3000;
+    const host = this.lbApp.restServer.config.host || '127.0.0.1';
+    this.server = this.app.listen(port, host);
+    await pEvent(this.server, 'listening');
+  }
+
+  public async stop() {
+    if (!this.server) return;
+    await this.lbApp.stop();
+    this.server.close();
+    await pEvent(this.server, 'close');
+    this.server = undefined;
+  }
+}
+```
+
+This app will replace the `CoffeeShopApplication` as the entry point for the
+program, modify the `src/index.ts` file accordingly.
+
+{% include code-caption.html content="src/index.ts" %}
+
+```ts
+import {ApplicationConfig} from '@loopback/core';
+import {ExpressServer} from './server';
+
+export {ExpressServer};
+
+export async function main(options: ApplicationConfig = {}) {
+  const server = new ExpressServer(options);
+  await server.boot();
+  await server.start();
+  console.log('Server is running at http://127.0.0.1:3000');
+}
+```
+
+Now create a static file named `express.html` to be served from the root of the
+server.
```

Comment:
I have mixed feelings about this part. It makes sense to have Express/root-level public files in the context of an example showing how to mount LB4 API to an Express app. Such approach does not make so much sense to me in the context of migrating a LB3 application to LB4.

What I think will be a better approach: explain users how to take their public files in LB3 app (typically `lb3app/client`) and expose them either via the top-level Express app or via LB4.

Ideally, the instructions should be generic and explain how to migrate any public files (mounted via `express.static`), not only the default LB3 setup (`client` directory in the LB3 project).

Reply:
Added "Move any static files from the LB3 app to the `public` directory of the Express
app.", and removed irrelevant instruction.